pkg/controller/certificates/metrics/controller.go

```diff
@@ -77,12 +80,13 @@ func NewController(
 	return &controller{
 		certificateLister: certificateInformer.Lister(),
 		metrics:           metrics,
+		contextTimeout:    contextTimeout,
 	}, queue, mustSync
 }
 
 func (c *controller) ProcessItem(ctx context.Context, key string) error {
 	// Set context deadline for full sync in 10 seconds
-	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
+	ctx, cancel := context.WithTimeout(ctx, c.contextTimeout)
```

Comment:
question: (as in my other comment) Do you have a reason you're wanting to configure the timeout here? To fix the linked issues I'd expect that we'd only want to configure `pkg/controller/clusterissuers/sync.go` and `pkg/controller/issuers/sync.go`.